src/core/components/stats/bw.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const Big = require('bignumber.js')
+const human = require('human-to-milliseconds')
+const errCode = require('err-code')
+
+function getBandwidthStats (libp2p, opts) {
+  let stats
+
+  if (opts.peer) {
+    stats = libp2p.metrics.forPeer(opts.peer)
+  } else if (opts.proto) {
+    stats = libp2p.metrics.forProtocol(opts.proto)
+  } else {
+    stats = libp2p.metrics.global
+  }
+
+  if (!stats) {
+    return {
+      totalIn: new Big(0),
+      totalOut: new Big(0),
+      rateIn: new Big(0),
+      rateOut: new Big(0)
+    }
+  }
+
+  const { movingAverages, snapshot } = stats
+
+  return {
+    totalIn: snapshot.dataReceived,
+    totalOut: snapshot.dataSent,
+    rateIn: new Big(movingAverages.dataReceived[60000].movingAverage() / 60),
+    rateOut: new Big(movingAverages.dataSent[60000].movingAverage() / 60)
+  }
+}
+
+module.exports = ({ libp2p }) => {
+  return async function * (options) {
+    options = options || {}
+
+    if (!options.poll) {
```

Comment:
Having a `poll` option feels slightly strange to me. I wonder if it would make sense to have two different methods:
- `stats()` that returns a single value
- `poll()` that is async iterable

Reply:
I agree...

I'm not sold on the suggestion yet...`ipfs.stats.bw.stats()` doesn't read well to me! It would be nice to have a convention for APIs where we want to expose a one shot request/response but also a live streaming version...

```js
// Single shot:
const stat = await ipfs.stats.bw()

// Live feed:
for await (const stat of ipfs.stats.bw.feed()) {
}
```

Reply:
ha yes that sounds much better :)